curvefs/src/mds/topology/topology_manager.cpp

```diff
@@ -721,6 +721,13 @@ bool TopologyManager::CreateCopysetNodeOnMetaServer(
     return true;
 }
 
+void TopologyManager::ClearCopysetCreating(PoolIdType poolId,
+    const std::set<CopySetIdType> &copysets) {
+    for (const auto &id : copysets) {
+        topology_->RemoveCopySetCreating(CopySetKey(poolId, id));
+    }
+}
+
 TopoStatusCode TopologyManager::CreateCopyset() {
```

Comment:
Does CreateCopyset have concurrency?

Reply:
> Does CreateCopyset have concurrency?

Concurrency occurs in extreme cases, but the copysets' id must be different.